2.1-Segwit-Version-1.ipynb

```diff
@@ -0,0 +1,564 @@
+    "\n",
+    "while (output.scriptPubKey != CScript([OP_1, program])):\n",
+    "    output = next(outputs)\n",
+    "    output_index += 1\n",
+    "output_value = output.nValue\n",
+    "        \n",
+    "# Set the output index for the spending tx input\n",
+    "print(\"Segwit v1 output index:\", output_index)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 3. Constructing a TX to spend the Segwit v1 output\n",
+    "\n",
+    "In order to contruct, sign and broadcast our transaction which spends our previous Segwit v1 output, we need to populate a `CTransaction` object.\n",
+    "\n",
+    "Overview of CTransaction members:\n",
+    " * `nVersion`\n",
+    " * `nLocktime`  \n",
+    " * `tx_vin` (list of `CTxIn` objects)\n",
+    " * `tx_vout` (list of `CTxOut` objects)\n",
+    " * `tx.wit.vtxinwit` (list of `CTxInWitness` objects)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.5: Construct CTransaction and populate inputs\n",
+    "\n",
+    "Populate:\n",
+    " * `nVersion`\n",
+    " * `nLocktime`\n",
+    " * `tx_vin` (list of `CTxIn` objects)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Construct schnorr transaction\n",
+    "spending_tx = CTransaction()\n",
+    "spending_tx.nVersion = 1\n",
+    "spending_tx.nLockTime = 0\n",
+    "outpoint = COutPoint(tx.sha256, output_index)\n",
+    "spending_tx_in = CTxIn(outpoint = outpoint)\n",
+    "spending_tx.vin = [spending_tx_in]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.6: Populate outputs of spending transaction\n",
+    "\n",
+    "* Generate a new wallet address to send funds to.\n",
+    "* Determine the fee amount.\n",
+    "* Populate the transaction output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate new Bitcoin Core wallet address\n",
+    "dest_addr = test.nodes[0].getnewaddress(address_type=\"bech32\")\n",
+    "scriptpubkey = bytes.fromhex(test.nodes[0].getaddressinfo(dest_addr)['scriptPubKey'])\n",
+    "\n",
+    "# Determine minimum fee required for mempool acceptance\n",
+    "min_fee = int(test.nodes[0].getmempoolinfo()['mempoolminfee'] * 100000000)\n",
+    "\n",
+    "# Complete output which returns funds to Bitcoin Core wallet\n",
+    "amount_sat = int(amount_btc * 100000000)\n",
+    "dest_output = CTxOut(nValue=amount_sat-min_fee, scriptPubKey=scriptpubkey)\n",
+    "spending_tx.vout = [dest_output]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.7: Sign the transaction with a Schnorr signature\n",
+    "\n",
+    "The current Taproot BIP proposes the following sighash flags:\n",
+    "* Legacy sighash flags: `[0x01,0x02,0x03,0x81,0x82,0x83]`\n",
```

Comment:
Adding the names of the sighashes and the hex in parentheses? (I honestly don't remember the numbers)

Reply:
Agreed this would be useful

Reply:
done